modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SpinnerSkinTest.java

```diff
@@ -149,4 +158,23 @@ public void testSpinnerSkin() {
         spinner.setSkin(new SpinnerSkin<>(spinner));
         spinner.setSkin(new SpinnerSkin<>(spinner));
     }
+
+    @Test
+    public void testSpinnerIncrementOnRemovingFromScene() {
+        spinner = new Spinner<>(0, 1000, 0);
+        spinner.setSkin(new SpinnerSkin<>(spinner));
+
+        HBox root = new HBox(spinner);
+        Scene scene = new Scene(root);
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.show();
+
+        SpinnerBehavior behavior = SpinnerSkinShim.getSpinnerBehavior((SpinnerSkin)spinner.getSkin());
+        behavior.startSpinning(true);
+
+        assertEquals(Status.RUNNING, SpinnerBehaviorShim.getTimeline(behavior).getStatus());
+        root.getChildren().clear();
+        assertEquals(Status.STOPPED, SpinnerBehaviorShim.getTimeline(behavior).getStatus());
```

Comment:
just curious: if we add the spinner back to the scene, should it spin again?  do we want  to add this condition to this test?

Reply:
It should not spin again. Added the same condition in the test. Please check